src/buildenv/buildenv.cc

```diff
@@ -114,7 +114,7 @@ static void addPkg(const Path & pkgDir, int priority)
                 return;
             throw SysError(format("opening ‘%1%’") % propagatedFN);
         }
-        propagated = readLine(fd.get());
+        propagated = readFile(fd.get());
     }
     for (const auto & p : tokenizeString<std::vector<string>>(propagated, " "))
```

Comment:
Now it only tokenizes on spaces right? Dropping the `" "` will cause it to split on any whitespace.